tasks/fullStack/app/controllers/reviews_controller.rb

```diff
@@ -1,44 +1,60 @@
 class ReviewsController < ApplicationController
+  protect_from_forgery except: :fetch_reviews
+
 
   DEFAULT_TAGS = ['default']
 
   def index
-    if params[:shop_id].present? && Shop.where("id = #{params[:shop_id]}").present?
-      params[:per_page] ||= 10
-      offset = params[:page].to_i * params[:per_page]
-
-      @data = []
-      products = Product.where("shop_id = #{params[:shop_id]}").sort_by(&:created_at)[offset..(offset + params[:per_page])]
-      products.each do |product|
-        reviews = product.reviews.sort_by(&:created_at)[offset..(offset + params[:per_page])]
-        @data << { product: product, reviews: reviews }
-      end
+    @products = Product.where(shop_id: params[:shop_id]).page params[:page]
+  end
+
+  def fetch_reviews
+    product = Product.includes(:reviews).find(params[:product_id])
+    @reviews = product.reviews.order(created_at: :desc).page params[:page]
```

Comment:
We shouldn't use `order(created_at: :desc)` without the existing index on this column.
There are 2 possible solutions:
1) use order(id: :desc) , because they are similar (>90%), the id values will be increased along with the created_at.
2) create the new index on the column created_at. But We wouldn't recommend this solution because it's not good for the DB cost, performance...

Reply:
Didn't come to my mind your first suggestion but I think is a good idea. I just wanted to be sure to get the latest reviews first.

I think adding an index at `:created_at` would be a bit weird but I see your point and also why would cost if you have million of reviews and keep adding them. However, adding an index, normally shouldn't bring any issue, but again, I think it depends on how often your write into the table and other variables like how big is the table, etc.